src/main/java/org/kuali/kra/subaward/lookup/SubAwardLookupableHelperServiceImpl.java

```diff
@@ -75,7 +78,20 @@
             throw new RuntimeException(e);
         }
 
-        return returnResults;
+        Integer searchResultsLimit = LookupUtils.getSearchResultsLimit(SubAward.class);
+        List<SubAward> searchResultsReturn = new ArrayList<SubAward>();
+        int i = 0;

```

Comment:
Please eliminate for loop and index and use the following code:
if ( searchResultsLimit < returnResults.size()){
          searchResultsReturn = returnResults.subList(0, searchResultsLimit);
} else {
          searchResultsReturn = returnResults;
}